src/Workspaces/Core/Portable/Diagnostics/DiagnosticData.cs

```diff
@@ -336,13 +349,28 @@ public static DiagnosticData Create(Diagnostic diagnostic, Document document)
                                                 .WhereNotNull()
                                                 .ToReadOnlyCollection();
 
+            var additionalProperties = GetAdditionalProperties(document, diagnostic);
+
+            var documentPropertiesService = document.Services.GetService<DocumentPropertiesService>();
```

Comment:
we have a service that isn't an interface?

Reply:
Yes.